tiled/client/base.py

```diff
@@ -121,6 +205,54 @@ def formats(self):
         )
         return sorted(formats)
 
+    def update_metadata(self, metadata=None, specs=None):
+        """
+        EXPERIMENTAL: Update metadata.
+
+        This is subject to change or removal without notice
+
+        Parameters
+        ----------
+        metadata : dict, optional
+            User metadata. May be nested. Must contain only basic types
+            (e.g. numbers, strings, lists, dicts) that are JSON-serializable.
+        specs : List[str], optional
+            List of names that are used to label that the data and/or metadata
+            conform to some named standard specification.
+        """
+
+        self._cached_len = None
+
+        data = {
+            "metadata": metadata,
+            "specs": specs,
+        }
+
+        full_path_meta = (
+            "/node/metadata"
+            + "".join(f"/{part}" for part in self.context.path_parts)
+            + "".join(f"/{part}" for part in (self._path or [""]))
+        )
+
+        content = self.context.put_json(full_path_meta, data)
+
+        if metadata is not None:
+            if len(content["metadata"]) > 0:
```

Comment:
If we make it here, we know the client submitted a change that was accepted. The only additional question we have now is, "Did the server modify the `metadata` (via a validator) before accepting it?" This should be answer by `content["metadata"] is None`, not be an `len()` check.